[improve][io] JDBC Sink: add flag to exclude non declared fields (#18008)
apache/pulsar: site2/docs/io-jdbc-sink.md

| `insertMode` | enum( INSERT,UPSERT,UPDATE) | false    | INSERT             | If it is configured as UPSERT, the sink uses upsert semantics rather than plain INSERT/UPDATE statements. Upsert semantics refer to atomically adding a new row or updating the existing row if there is a primary key constraint violation, which provides idempotence.                                                     |
| `nullValueAction` | enum(FAIL, DELETE) | false    | FAIL               | How to handle records with NULL values. Possible options are `DELETE` or `FAIL`.                                                                                                                                                                                                                                             |
| `useTransactions` | boolean | false    | true               | Enable transactions of the database.                                                                                                                                                                                                                                                                                         

> | `excludeNonDeclaredFields` | boolean | false    | false              | All the table fields are discovered automatically. This fields indicate if the fields not explicitly listed in `nonKey` and `key` must be included in the query. By default all the fields of the table are included. In order to leverage of fields defaults during insertion, it is suggested to set this value to `false`. |

Reviewer: ```suggestion
| `excludeNonDeclaredFields` | boolean | false    | false              | All the table fields are discovered automatically. These fields indicate if the fields are not explicitly listed in `nonKey` and `key`, they must be included in the query. By default, all the fields of the table are included. To leverage the default values of fields during insertion, it is suggested to set this value to `false`. |
```
Do you mean this?

Author: Nope but thanks, I updated the doc
